dart/lib/src/sentry_options.dart

```diff
@@ -242,8 +243,10 @@ class SentryOptions {
 
 /// This function is called with an SDK specific event object and can return a modified event
 /// object or nothing to skip reporting the event
-typedef BeforeSendCallback = SentryEvent? Function(SentryEvent event,
-    {dynamic hint});
+typedef BeforeSendCallback = FutureOr<SentryEvent?> Function(
```

Comment:
should we update the tests in sentry_client_test?

Reply:
They're still valid and green, so I'm not quite sure what you're trying to say.

Reply:
while they don't break because of `FutureOr` which is a good thing, we could either add a new test with the new signature (having an async callback), or make the current ones async., I'm fine either way.